src/controllers/plugins/LuaUtilities.hpp

```diff
@@ -54,18 +65,111 @@ StackIdx push(lua_State *L, const std::string &str);
 StackIdx push(lua_State *L, const bool &b);
 
 // returns OK?
+bool peek(lua_State *L, bool *out, StackIdx idx = -1);
 bool peek(lua_State *L, double *out, StackIdx idx = -1);
 bool peek(lua_State *L, QString *out, StackIdx idx = -1);
 bool peek(lua_State *L, QByteArray *out, StackIdx idx = -1);
 bool peek(lua_State *L, std::string *out, StackIdx idx = -1);
+bool peek(lua_State *L, api::CompletionList *out, StackIdx idx = -1);
 
 /**
  * @brief Converts Lua object at stack index idx to a string.
  */
 QString toString(lua_State *L, StackIdx idx = -1);
 
+// Special
+class BalanceKepper
```

Comment:
Add a comment to this, and probably rename to `BalanceKeeper`, maybe something including `Stack`?